src/commands/resetIndex.js

```diff
@@ -0,0 +1,65 @@
+import path from 'path'
+
+import { GitIndexManager } from '../managers/GitIndexManager.js'
+import { GitObjectManager } from '../managers/GitObjectManager.js'
+import { FileSystem } from '../models/FileSystem.js'
+
+import { readObject } from './readObject'
+import { log } from './log'
+
+/**
+ * Reset a file from the git index (aka staging area)
+ *
+ * @link https://isomorphic-git.github.io/docs/resetIndex.html
+ */
+export async function resetIndex ({
+  dir,
+  gitdir = path.join(dir, '.git'),
+  fs: _fs,
+  filepath,
+  ref = 'HEAD'
+}) {
+  try {
+    const fs = new FileSystem(_fs)
+    const tree = await log({ gitdir, fs, ref, depth: 1 })
```

Comment:
Can you use `src/utils/resolveTree.js` here instead of `log`? `log` will only resolve commits, whereas `resolveTree` will also handle raw OIDs or tags that point directly to `tree` objects.

Reply:
I have only `filepath` and `ref` but `src/utils/resolveTree.js` takes only `oid`

Reply:
the `log` command can be replaced with this
```
let oid = await GitRefManager.resolve({ fs, gitdir, ref })
const tree = await logCommit({ fs, gitdir, oid })
oid = tree && tree.tree
```

Reply:
Actually... you just need the first line. `readObject` uses `resolveTree` internally so all you need is the `oid` of the commit, you don't need to do `logCommit` and get the tree.